docs/design/velero-krm-enhancement.md

```diff
@@ -0,0 +1,64 @@
+# Velero KRM Plugin Design
+
+### Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] User-facing documentation is created
+
+### Open Questions 
+-  Can changes be made in exising codebase as opposed to refactoring the whole code? 
+-  Whether adding KRM (Kubernetes Resource Management) functions is actually useful or will it affect the performance of current plugin in a negative manner?
+
+
+### Summary
+
+Currently Velero plugin framework is inefficient. Velero server executes 1 gRPC call per plugin, meaning 1 gRPC call per resource transformation.  In order to overcome this we introduce a new plugin type that will allow Velero to leverage KRM functions for resource transformation and additional item gathering so that we can drastically improve the plugin framework performance by processing N items with a single gRPC call.
+
+### Motivation
+
+- To have a common set of KRM functions which are reusable.
+- Improving the performance of Velero backup-restore process by only having one grpc call per plugin when processing backup/restore of a resource.
+and passing a list of resources as opposed to having one grpc call to backup/restore a resource. 
+
+### Goals
+- Check and research on if the given approach is feasible.
+- Make changes to OpenShift Velero Plugins and introduce a new plugin type to pass all resources and for  implementing KRM functions
+- Modify the existing OADP controller code to pass all resource list to velero plugin backup/restore interface.
+
+
+### User Stories 
+- User will be developing a new plugin type that can handle multiple resources for common transformations
```

Comment:
Maybe the resources which can be transformed ? or a concrete example about one resource?

Reply:
yeah an example about a resource would be good to understand the usecase

Reply:
we are handling multiple resources though right?

I think that the example may just need to point to the ResourceList type as what it would handle.

I think the user story should be something like 

```
1. As a user of velero, I would like to re-use existing commuity-built KRM functions as velero plugins
2. As a developer of velero plugins, I would like to increase the throughput of a plugin that handles a lot of resources, by getting them all at once.
```